Description: Fix build failures with standard C 2023.
 This patch adds missing function argument types in prototypes, and
 casts a couple of indirect function calls, to ensure compatibility with
 the standard C 2023, now selected by default by gcc 15 and later.
Author: Étienne Mollier <emollier@debian.org>
Bug-Debian: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1097250
Forwarded: https://github.com/PDLPorters/perl5-PGPLOT/pull/18
Last-Update: 2025-02-25
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/pgfun.c
+++ b/pgfun.c
@@ -26,9 +26,9 @@
 /* Prototypes */
 
 static SV*  pgfunname[2];
-float pgfun1();
-float pgfun2();
-void   pgfunplot();
+float pgfun1(float *x);
+float pgfun2(float *x);
+void   pgfunplot(int *visible,float *x, float *y, float *z);
 void PGFUNX(float fy(), int *n, float *xmin, float *xmax, int *pgflag);
 void PGFUNY(float fx(), int *n, float *ymin, float *ymax, int *pgflag);
 void PGFUNT(float fx(), float fy(), int *n, float *tmin, float *tmax, int *pgflag);
--- a/PGPLOT.xs
+++ b/PGPLOT.xs
@@ -306,7 +306,7 @@
   CODE:
     DEBUG_PRINT(pgconx);
     pgfunname[0] = plot;
-    cpgconx(a,idim,jdim,i1,i2,j1,j2,c,nc,pgfunplot);
+    cpgconx(a,idim,jdim,i1,i2,j1,j2,c,nc,(void(*)(void))pgfunplot);
 
 
 void
@@ -462,7 +462,7 @@
     DEBUG_PRINT(pgfunt);
     pgfunname[0] = fx;
     pgfunname[1] = fy;
-    cpgfunt(pgfun1,pgfun2,n,tmin,tmax,pgflag);
+    cpgfunt((float(*)(void))pgfun1,(float(*)(void))pgfun2,n,tmin,tmax,pgflag);
 
 
 void
@@ -475,7 +475,7 @@
   CODE:
     DEBUG_PRINT(pgfunx);
     pgfunname[0] = fy;
-    cpgfunx(pgfun1,n,xmin,xmax,pgflag);
+    cpgfunx((float(*)(void))pgfun1,n,xmin,xmax,pgflag);
 
 
 void
@@ -488,7 +488,7 @@
   CODE:
     DEBUG_PRINT(pgfuny);
     pgfunname[0] = fx;
-    cpgfuny(pgfun1,n,ymin,ymax,pgflag);
+    cpgfuny((float(*)(void))pgfun1,n,ymin,ymax,pgflag);
 
 
 void
